src/main.c

```diff
@@ -43,6 +50,10 @@ static void setup_clock(void)
 
 	/* ADC */
 	rcc_periph_clock_enable(RCC_ADC1);
+
+	/* TIM1 */
+	rcc_periph_clock_enable(RCC_TIM1);
+
```

Comment:
I would avoid this newline before the closing `}` just for consistency (not used in any other function).

Reply:
fixed with clang-format ^^